actors/builtin/market/market_test.go

```diff
@@ -1709,6 +1711,181 @@ func TestMarketActorDeals(t *testing.T) {
+
+		rt.ExpectComputeUnsealedSectorCID(1, []abi.PieceInfo{pi}, cid.Cid{}, errors.New("error"))
+		rt.SetCaller(provider, builtin.StorageMinerActorCodeID)
+		rt.ExpectValidateCallerType(builtin.StorageMinerActorCodeID)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			rt.Call(actor.ComputeDataCommitment, param)
+		})
+	})
+}
+
+func TestVerifyDealsForActivation(t *testing.T) {
+	owner := tutil.NewIDAddr(t, 101)
+	provider := tutil.NewIDAddr(t, 102)
+	worker := tutil.NewIDAddr(t, 103)
+	client := tutil.NewIDAddr(t, 104)
+	mAddrs := &minerAddrs{owner, worker, provider}
+	sectorStart := abi.ChainEpoch(1)
+	sectorExpiry := abi.ChainEpoch(200)
+	start := abi.ChainEpoch(10)
+	end := abi.ChainEpoch(20)
+
+	t.Run("verify deal and get deal weight for unverified deal proposal", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		dealId := actor.generateAndPublishDeal(rt, client, mAddrs, start, end)
+		d := actor.getDealProposal(rt, dealId)
+
+		resp := actor.verifyDealsForActivation(rt, provider, sectorStart, sectorExpiry, dealId)
+		require.EqualValues(t, big.Zero(), resp.VerifiedDealWeight)
+		require.EqualValues(t, market.DealWeight(d), resp.DealWeight)
+	})
+
+	t.Run("verify deal and get deal weight for verified deal proposal", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		deal := actor.generateDealAndAddFunds(rt, client, mAddrs, start, end)
+		deal.VerifiedDeal = true
+		dealIds := actor.publishDeals(rt, mAddrs, deal)
+
+		resp := actor.verifyDealsForActivation(rt, provider, sectorStart, sectorExpiry, dealIds...)
+		require.EqualValues(t, market.DealWeight(&deal), resp.VerifiedDealWeight)
+		require.EqualValues(t, big.Zero(), resp.DealWeight)
+	})
+
+	t.Run("verification and weights for verified and unverified deals", func(T *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+
+		vd1 := actor.generateDealAndAddFunds(rt, client, mAddrs, start, end)
+		vd1.VerifiedDeal = true
+
+		vd2 := actor.generateDealAndAddFunds(rt, client, mAddrs, start, end+1)
+		vd2.VerifiedDeal = true
+
+		d1 := actor.generateDealAndAddFunds(rt, client, mAddrs, start, end+2)
+		d2 := actor.generateDealAndAddFunds(rt, client, mAddrs, start, end+3)
+
+		dealIds := actor.publishDeals(rt, mAddrs, vd1, vd2, d1, d2)
+
+		resp := actor.verifyDealsForActivation(rt, provider, sectorStart, sectorExpiry, dealIds...)
+
+		verifiedWeight := big.Add(market.DealWeight(&vd1), market.DealWeight(&vd2))
+		nvweight := big.Add(market.DealWeight(&d1), market.DealWeight(&d2))
+		require.EqualValues(t, verifiedWeight, resp.VerifiedDealWeight)
+		require.EqualValues(t, nvweight, resp.DealWeight)
+	})
+
+	t.Run("fail when caller is not a StorageMinerActor", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		param := &market.VerifyDealsForActivationParams{SectorStart: sectorStart, SectorExpiry: sectorExpiry}
```

Comment:
This call fails for multiple reasons, e.g. deal not found. In this case the exit code probably distinguishes them, but the test would be stronger if the caller was the _only_ thing wrong with it. In other tests below, the exit code is the same, so you can't actually tell if the condition you think is causing the test to fail is the right one (it might be masked by something else).

I expect them to all look like the basic success case above, calling `actor.verifyDealsForActivation` with a good deal

Reply:
@anorth Have fixed this.